### PR TITLE
Add Cluster object codegen support

### PR DESCRIPTION
**Pull request by yunhanw-google**

### CODE REVIEW

File: zzz_generated/all-clusters-app/zap-generated/cluster-objects.h [hunk 1]
```diff
@@ -0,0 +1,1305 @@
+/*
```
Comment (mrjerryjohns) [inline zzz_generated/all-clusters-app/zap-generated/cluster-objects.h]: We don't need to generate a unique `cluster-objects.cpp`, and `.h` file for each app. In fact, the whole goal of this cluster object model is to have just 1 version across all apps :) 

That version should be in `zzz_generated/common/zap-generated/`

File: zzz_generated/all-clusters-app/zap-generated/cluster-objects.h [hunk 2]
```diff
@@ -0,0 +1,1305 @@
...
+    EndDevice       = 0x03,
+    REED            = 0x04,
+    Router          = 0x05,
+    Leader          = 0x06,
+};
+
+namespace NeighborTable {
+enum FieldId
+{
+    kExtAddressFieldId = 0; kAgeFieldId = 1; kRloc16FieldId = 2; kLinkFrameCounterFieldId = 3; kMleFrameCounterFieldId = 4;
```
Comment (mrjerryjohns) [inline zzz_generated/all-clusters-app/zap-generated/cluster-objects.h]: Formatting issue?

File: zzz_generated/all-clusters-app/zap-generated/cluster-objects.h [hunk 3]
```diff
@@ -0,0 +1,1305 @@
...
+{
+    kFabricIndexFieldId = 0; kRootPublicKeyFieldId = 1; kVendorIdFieldId = 2; kFabricIdFieldId = 3; kNodeIdFieldId = 4;
+    kLabelFieldId                                                                                                  = 5;
+};
+struct Type
+{
+    uint8_t FabricIndex;
+    chip::ByteSpan RootPublicKey;
+    uint16_t VendorId;
+    FABRIC_ID FabricId;
```
Comment (mrjerryjohns) [inline zzz_generated/all-clusters-app/zap-generated/cluster-objects.h]: Right type?

File: zzz_generated/all-clusters-app/zap-generated/cluster-objects.h [hunk 4]
```diff
@@ -0,0 +1,1305 @@
...
+    NODE_ID NodeId;
```
Comment (mrjerryjohns) [inline zzz_generated/all-clusters-app/zap-generated/cluster-objects.h]: Right type?

**Comment by mrjerryjohns** [comment]
Couple of key items missing:
* Needs to delete the 'fake' clusters-objects I've checked into `src/app/common`
* The BUILD.gn in that folder needs to be updated to pick up the **single** cluster-objects.cpp in zzz-generated/app-common/zap-generated/cluster-objects.cpp` (which is where you'll code-gen to).
* We need to decide if that BUILD.gn should automatically include the `zzz-generated/app-common/zap-generated/` as part of the include path, so that consumers using cluster objects can just do:

```
#include <cluster-objects.h>
```

And not worry about the actual location of the cluster objects.

**Comment by mrjerryjohns** [comment]
We also then need to add that BUILD.gn as a dependency for all example apps today to ensure it gets built.

File: zzz_generated/all-clusters-app/zap-generated/cluster-objects.h (hunk 1 above)
Comment (yunhanw-google) [reply zzz_generated/all-clusters-app/zap-generated/cluster-objects.h]: Done, thanks

File: zzz_generated/all-clusters-app/zap-generated/cluster-objects.h (hunk 2 above)
Comment (yunhanw-google) [reply zzz_generated/all-clusters-app/zap-generated/cluster-objects.h]: Done, thanks

File: zzz_generated/all-clusters-app/zap-generated/cluster-objects.h (hunk 4 above)
Comment (yunhanw-google) [reply zzz_generated/all-clusters-app/zap-generated/cluster-objects.h]: Fixed, thanks

File: zzz_generated/all-clusters-app/zap-generated/cluster-objects.h (hunk 3 above)
Comment (yunhanw-google) [reply zzz_generated/all-clusters-app/zap-generated/cluster-objects.h]: fixed, thanks

File: zzz_generated/app-common/app-common/zap-generated/cluster-objects.h [hunk 5]
```diff
@@ -0,0 +1,1507 @@
...
+};
+struct Type
+{
+    DataModel::List<NestedStructList::Type> a;
+    CHIP_ERROR Encode(TLV::TLVWriter & writer, uint64_t tag) const;
+    CHIP_ERROR Decode(TLV::TLVReader & reader);
+};
+struct DecodableType
+{
+    DataModel::DecodableList<NestedStructList::Type> a;
```
Comment (yunhanw-google) [inline zzz_generated/app-common/app-common/zap-generated/cluster-objects.h]: @mrjerryjohns  this line stucks....NestedStructList::Type ==> NestedStructList::DecodableType

File: zzz_generated/app-common/app-common/zap-generated/cluster-objects.cpp [hunk 6]
```diff
@@ -0,0 +1,1027 @@
...
+            break;
+        default:
+            break;
+        }
+    }
+    VerifyOrReturnError(err == CHIP_END_OF_TLV, err);
+    ReturnErrorOnFailure(reader.ExitContainer(outer));
+    return CHIP_NO_ERROR;
+}
+} // namespace ContentLaunchBrandingInformation
```
Comment (mrjerryjohns) [inline zzz_generated/app-common/app-common/zap-generated/cluster-objects.cpp:234]: Add at least 1, preferably 2 newlines between these namespaces.

**Pull request closed without merging**